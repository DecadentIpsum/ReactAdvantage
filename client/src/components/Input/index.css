.input {
  width: 100%; }

.input-wrapper {
  position: relative; }

.input-error {
  position: absolute;
  top: calc(100% + 2px);
  color: #D9534F;
  font-size: 0.8rem; }
