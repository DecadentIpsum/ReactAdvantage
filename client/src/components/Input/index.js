import React, { Component } from 'react';
import { InputText } from 'primereact/components/inputtext/InputText';
import { InputMask } from 'primereact/components/inputmask/InputMask';
import { Password } from 'primereact/components/password/Password';
import './index.css';

export default class Input extends Component {
    constructor(props) {
        super(props);

        let Input = InputText;

        if (this.props.mask) {
            Input = InputMask;
        } else if (this.props.password) {
            Input = Password;
        }

        this.InputType = Input;
    }

    render() {
        const {
            className = '',
            error,
            password,
            ...restProps
        } = this.props;

        const { InputType } = this;

        const classNames = ['input'].concat(className.split(' '));

        error && classNames.push('ui-state-error');

        return (
            <div className="input-wrapper">
                <InputType
                    className={classNames}
                    {...restProps}
                />
                {error && <div className="input-error">{error}</div>}
            </div>
        );
    }
}
