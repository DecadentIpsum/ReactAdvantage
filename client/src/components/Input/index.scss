.input {
    width: 100%;
}

.input-wrapper {
    position: relative;
}

.input-error {
    position: absolute;
    top: calc(100% + 2px);
    color: #f44336;
    font-size: 0.8rem;
}