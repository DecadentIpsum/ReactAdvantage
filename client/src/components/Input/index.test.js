import React from 'react';
import Enzyme, { shallow } from 'enzyme';

import Input from './index';

it('renders correctly', () => {
    const component = shallow(<Input />);

    expect(component).toMatchSnapshot();
});