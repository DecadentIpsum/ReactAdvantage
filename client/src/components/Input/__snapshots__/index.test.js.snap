// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Input />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <InputText
          className={
                    Array [
                              "input",
                              "",
                            ]
          }
          keyfilter={null}
          onInput={null}
          onKeyPress={null}
          validateOnly={false}
/>,
        undefined,
      ],
      "className": "input-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": Array [
            "input",
            "",
          ],
          "keyfilter": null,
          "onInput": null,
          "onKeyPress": null,
          "validateOnly": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <InputText
            className={
                        Array [
                                    "input",
                                    "",
                                  ]
            }
            keyfilter={null}
            onInput={null}
            onKeyPress={null}
            validateOnly={false}
/>,
          undefined,
        ],
        "className": "input-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": Array [
              "input",
              "",
            ],
            "keyfilter": null,
            "onInput": null,
            "onKeyPress": null,
            "validateOnly": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
